src/java/org/apache/cassandra/service/reads/thresholds/CoordinatorWarnings.java

```diff
@@ -147,14 +152,14 @@ private static void clearState()
     @FunctionalInterface
     private interface ToString
     {
-        String apply(int count, long value, String cql);
+        String apply(ImmutableSet<InetAddressAndPort> nodes, long value, String cql);
```

Comment:
It is cool to know what exact replicas violated the tresholds so an operator can e.g. try to compact them to lower the amount of referenced indexes.

Reply:
I mean, you could leave it, but then you a.) have change the message in `tombstoneAbortMessage()` et al. and b.) might have to deal w/ printing an arbitrarily large set of nodes, right? (You could like, truncate at 3 nodes or something...)

I'm guessing the original intent (CC @dcapwell) was to let users see the problem in aggregate and then hit the logs to see what nodes exactly were having issues. I'd vote for either leaving things as they are or printing a truncated list of problematic nodes in the implementations of `ToString`.

Reply:
Rather to keep it as it was before then truncating it. I dont like to truncate it. BTW I dont think it would be too long anyway ... what RF you have out there at most? 5? 9? And not all replicas are going to violate ... it is not like this list would be practically without a reasonable limit.

Reply:
This could be hundreds of nodes when the partition key is not restricted, i.e. not RF, but N/RF

Also, when one node violates, it does become much more likely that others will shortly (if they aren't already).

Reply:
OK